stripe/resource.py

```diff
@@ -273,6 +273,22 @@ def __delitem__(self, k):
         if hasattr(self, '_unsaved_values'):
             self._unsaved_values.remove(k)
 
+    def __setstate__(self, state):
+        self.update(state)
+
+    def __reduce__(self):
```

Comment:
Can we do comments on both `__setstate__` and `__reduce__` that follow roughly exactly what you said in the PR description?

Reply:
(Someone reading this who didn't see the original PR might wonder what exactly is going on here.)